lib/dialects/db2/query-generator.js

```diff
@@ -0,0 +1,883 @@
+'use strict';
+
+const _ = require('lodash');
+const Utils = require('../../utils');
+const DataTypes = require('../../data-types');
+const AbstractQueryGenerator = require('../abstract/query-generator');
+const randomBytes = require('crypto').randomBytes;
+const Op = require('../../operators');
+
+/* istanbul ignore next */
+const throwMethodUndefined = function(methodName) {
+  throw new Error(`The method "${  methodName  }" is not defined! Please add it to your sql dialect.`);
+};
+
+class Db2QueryGenerator extends AbstractQueryGenerator {
+  constructor(options) {
+    super(options);
+
+    this.OperatorMap = Object.assign({}, this.OperatorMap, {
+      [Op.regexp]: 'REGEXP_LIKE',
+      [Op.notRegexp]: 'NOT REGEXP_LIKE'
+    });
+    this.autoGenValue = 1;
+  }
+
+  createSchema(schema) {
+    return [
+      'CREATE SCHEMA',
+      this.quoteIdentifier(schema),
+      ';'
+    ].join(' ');
+  }
+
+  dropSchema(schema) {
+    // DROP SCHEMA Can't drop schema if it is not empty.
+    // DROP SCHEMA Can't drop objects belonging to the schema
+    // So, call the admin procedure to drop schema.
+    const query = `CALL SYSPROC.ADMIN_DROP_SCHEMA(${ wrapSingleQuote(schema.trim()) }, NULL, ? , ?)`;
+    const sql = { query };
+    sql.bind = [{ ParamType: 'INOUT', Data: 'ERRORSCHEMA' },
+      { ParamType: 'INOUT', Data: 'ERRORTABLE' }];
+    return sql;
+  }
+
+  showSchemasQuery() {
+    return 'SELECT SCHEMANAME AS "schema_name" FROM SYSCAT.SCHEMATA WHERE ' +
+      "(SCHEMANAME NOT LIKE 'SYS%') AND SCHEMANAME NOT IN ('NULLID', 'SQLJ', 'ERRORSCHEMA')";
+  }
+
+  versionQuery() {
+    return 'select service_level as VERSION from TABLE (sysproc.env_get_inst_info()) as A';
+  }
+
+  createTableQuery(tableName, attributes, options) {
+    const query = 'CREATE TABLE <%= table %> (<%= attributes %>)',
+      primaryKeys = [],
+      foreignKeys = {},
+      attrStr = [],
+      commentTemplate = ' -- <%= comment %>, ' +
+          'TableName = <%= table %>, ColumnName = <%= column %>;';
+
+    let commentStr = '';
+
+    for (const attr in attributes) {
+      if (attributes.hasOwnProperty(attr)) {
```

Comment:
To fix linting:

```suggestion
      if (Object.prototype.hasOwnProperty.call(attributes, attr)) {
```

Reply:
@papb Thanks for the suggestion and review. Addressed the issues in latest commit.